### PR TITLE
`derive(ThinContext)`

### PR DESCRIPTION
**Pull request by mkatychev**
create `ThinContext` derive macro:
https://github.com/knox-networks/bigerror/blob/32c2de2f2593a2a416fbb144d237acce66d5d350/bigerror/src/context.rs#L31-L33
The above expands to:
https://github.com/knox-networks/bigerror/blob/4386c31b153baa9c6df153ad4b135583fd443551/bigerror_derive/src/context.rs#L36-L51
note: `#[bigerror(crate)]` is not needed when using `bigerror` as a dependency

* renamed `bigerror::error_stack::Result` to `bigerror:BigResult` to distinguish it from `std::result::Result`
* removed `from_report!` macro

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,76 +1,41 @@
...
-version = "0.9.0"
+[workspace]
+members = [
+    "bigerror", "bigerror_derive",
+    # "bigerror_dyn",
+]
+resolver = "2"
+
+[workspace.package]
+version = "0.9.0-alpha"
```
Comment (jsgoyette) [inline Cargo.toml:9]: Just adding the `alpha` decrements the version. Should the patch version also get updated? 
```suggestion
version = "0.9.1-alpha"
```

Comment (jsgoyette) [reply Cargo.toml:9]: Sorry, I was thinking about as if it were an RC. Just adding the `alpha` is ok.

File: bigerror_derive/src/attributes.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+fn try_set_attribute<T: ToTokens>(
+    attribute: &mut Option<T>,
+    value: T,
+    name: &'static str,
+) -> Result<(), Error> {
+    if attribute.is_some() {
+        return Err(Error::new_spanned(
+            value,
+            format!("{} already specified", name),
```
Comment (gshuflin) [inline bigerror_derive/src/attributes.rs:12]: nit: can be `{name} is already specified`

File: justfile [hunk 3]
```diff
@@ -8,6 +8,11 @@ fmt:
     rustup run nightly cargo fmt -- \
       --config-path ./fmt/rustfmt.toml
 
+[group("lint")]
+lint *args:
+    cd {{invocation_directory()}}; \
```
Comment (gshuflin) [inline justfile:13]: You can use the `[no-cd]` directive https://just.systems/man/en/chapter_34.html?highlight=no-cd#disabling-changing-directory190 here

File: Cargo.toml (hunk 1 above)
Comment (mkatychev) [reply Cargo.toml:9]: no worries, we could always increment it by `alpha.1` but I figure we'd do that in `0.9.0-beta.1`

**Review by jsgoyette — APPROVED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -1,76 +1,41 @@
-[package]
-name = "bigerror"
-version = "0.9.0"
+[workspace]
+members = [
+    "bigerror", "bigerror_derive",
+    # "bigerror_dyn",
```
Comment (mkatychev) [inline Cargo.toml:4]:
```suggestion
```

**Pull request merged**